Code/Framework/AzToolsFramework/Tests/FocusMode/EditorFocusModeTests.cpp

```diff
@@ -6,123 +6,90 @@
  *
  */
 
-#include <AzTest/AzTest.h>
-#include <AzCore/UserSettings/UserSettingsComponent.h>
-#include <AzCore/Component/TransformBus.h>
-#include <AzToolsFramework/FocusMode/FocusModeInterface.h>
-#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <Tests/FocusMode/EditorFocusModeFixture.h>
 
 namespace AzToolsFramework
 {
-    class EditorFocusModeTests
-        : public ::testing::Test
+    TEST_F(EditorFocusModeFixture, EditorFocusModeTests_SetFocus)
     {
-    protected:
-        void SetUp() override
-        {
-            m_app.Start(m_descriptor);
-
-            // Without this, the user settings component would attempt to save on finalize/shutdown. Since the file is
-            // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash 
-            // in the unit tests.
-            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+        // When an entity is set as the focus root, GetFocusRoot should return its EntityId.
+        m_focusModeInterface->SetFocusRoot(m_entityMap[CarEntityName]);
+        EXPECT_EQ(m_focusModeInterface->GetFocusRoot(), m_entityMap[CarEntityName]);
 
-            GenerateTestHierarchy();
-        }
+        // Calling ClearFocusRoot restores the default focus root (which is an invalid EntityId).
+        m_focusModeInterface->ClearFocusRoot();
+        EXPECT_EQ(m_focusModeInterface->GetFocusRoot(), AZ::EntityId());
+    }
 
-        void GenerateTestHierarchy() 
+    TEST_F(EditorFocusModeFixture, EditorFocusModeTests_IsInFocusSubTree_AncestorsDescendants)
+    {
+        // When the focus is set to an entity, all its descendants are in the focus subtree while the ancestors aren't.
         {
-            /*
-            *   City
-            *   |_  Street
-            *       |_  Car
-            *       |   |_ Passenger
-            *       |_  SportsCar
-            *           |_ Passenger
-            */
-
-            m_entityMap["cityId"] =         CreateEditorEntity("City",      AZ::EntityId());
-            m_entityMap["streetId"] =       CreateEditorEntity("Street",    m_entityMap["cityId"]);
-            m_entityMap["carId"] =          CreateEditorEntity("Car",       m_entityMap["streetId"]);
-            m_entityMap["passengerId1"] =   CreateEditorEntity("Passenger", m_entityMap["carId"]);
-            m_entityMap["sportsCarId"] =    CreateEditorEntity("SportsCar", m_entityMap["streetId"]);
-            m_entityMap["passengerId2"] =   CreateEditorEntity("Passenger", m_entityMap["sportsCarId"]);
+            m_focusModeInterface->SetFocusRoot(m_entityMap[StreetEntityName]);
+
+            EXPECT_EQ(m_focusModeInterface->IsInFocusSubTree(m_entityMap[CityEntityName]), false);
```

Comment:
suggestion: this could potentially be simplified with a test matcher to remove the duplication (this won't compile but something like.... `EXPECT_THAT(m_entityMap, Each(m_focusModeInterface->IsInFocusSubTree)` - it might not quite be that but things like that can be super useful - more info here - http://google.github.io/googletest/reference/matchers.html

Reply:
It would probably work fine if they were all supposed to be the same value, but since they are different (especially the later tests) I'm worries that trying to simplify the check to a one-liner may make it less readable?

I'll look into the page cause it's interesting, but I'm not sure I'll make the change if it looks more complex :)